lte/cloud/go/services/subscriberdb/obsidian/models/conversion.go

```diff
@@ -209,6 +220,17 @@ func getAssignedIPAddress(mobilitydState state.ArbitraryJSON) (string, error) {
 	return base64DecodeIPAddress(ipAddressAsString)
 }
 
+// GetIMSIAndAPNFromMobilitydStateKey returns the IMSI and APN from the
+// mobilityd state key of IMSI.APN.
+func GetIMSIAndAPNFromMobilitydStateKey(key string) (string, string, error) {
+	matches := imsiAPNRegex.FindStringSubmatch(key)
+	if len(matches) != imsiAPNExpectedMatchCount {
+		return "", "", errors.Errorf("mobilityd state key %s did not match regex", key)
+	}
+	imsi, apn := matches[1], matches[2]
+	return imsi, apn, nil
+}
+
```

Comment:
these functions probably belong in a different package/file now, we shouldn't have clients calling into this package except to instantiate and validate/convert models